src/LibExpat.jl

```diff
@@ -265,27 +232,57 @@ end
 # SetBase and GetBase
 
 
-
-function xp_parse(txt::AbstractString)
-    xph = nothing
-    xph = xp_make_parser()
-
+function xp_parse(xph::XPHandle, txt::AbstractString)
     try
         rc = XML_Parse(xph.parser, txt, sizeof(txt), 1)
         (rc != XML_STATUS_OK) && error("Error parsing document : $rc")
 
-        # The root element will only have a single child element in a well formed XML
-        return xph.pdata.elements[1]
+        return parsed_data(xph)
     catch e
         stre = string(e)
         (err, line, column, pos) = xp_geterror(xph)
         rethrow("$e, $err, $line, $column, $pos")
 
     finally
-        (xph != nothing) && xp_close(xph)
+        xp_close(xph)
     end
 end
 
+function xp_parse(p::XML_Parser, txt::AbstractString)
```

Comment:
this would be a much smaller diff if you avoided moving this function around

Reply:
I moved it, but not sure it's changing a lot, xp_parse function was in different places.
Thanks for review.